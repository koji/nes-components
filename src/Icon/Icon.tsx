import * as React from 'react';
import '../assets/nes.css';

type Props = {
  iconType: 'close' | 'trophy' | 'coin';
  iconSize: 'large' | 'medium' | 'small';
};

export class NesIcon extends React.Component<Props> {
  render() {
    const { iconType, iconSize } = this.props;
    const iconClass = `nes-icon ${iconType} is-${iconSize}`;
    return <i className={iconClass}></i>;
  }
}
