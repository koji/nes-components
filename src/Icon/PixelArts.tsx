import * as React from 'react';
import '../../node_modules/nes.css/css/nes.css';

type Props = {
  iconType:
    | 'nes-logo'
    | 'nes-jp-logo'
    | 'snes-logo'
    | 'snes-jp-logo'
    | 'nes-octocat'
    | 'nes-smartphone'
    | 'nes-phone';
  animate?: boolean; // this is only for nes-octocat
};

export class NesArt extends React.Component<Props> {
  render() {
    const { iconType, animate } = this.props;
    return <i className={animate ? `${iconType} animate` : `${iconType}`}></i>;
  }
}
