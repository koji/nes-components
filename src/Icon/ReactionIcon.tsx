import * as React from "react";
import "../../node_modules/nes.css/css/nes.css";

type Props = {
  iconType: "heart" | "star" | "like";
  iconSize: "large" | "medium" | "small";
  iconReaction?: "empty" | "half" | "transparent"; // undefined -> normal
};

export class NesReactionIcon extends React.Component<Props> {
  render() {
    const { iconType, iconSize, iconReaction } = this.props;
    const iconClass = `nes-icon is-${iconSize} ${iconType} is-${iconReaction}`;
    return <i className={iconClass}></i>;
  }
}
