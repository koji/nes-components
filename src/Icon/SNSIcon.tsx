import * as React from "react";
import "../../node_modules/nes.css/css/nes.css";

// snsType
// twitter, facebook, instagram, github, google
// gmail medium linkedin, twitch, youtube, reddit, whatsapp

// icon size
// large, medium, small

type Props = {
  snsType:
    | "twitter"
    | "facebook"
    | "instagram"
    | "github"
    | "google"
    | "gmail"
    | "medium"
    | "linkedin"
    | "twitch"
    | "youtube"
    | "reddit"
    | "whatsapp";
  iconSize: "large" | "medium" | "small";
};

export class NesSNSIcon extends React.Component<Props> {
  render() {
    const { snsType, iconSize } = this.props;
    const classType = `nes-icon ${snsType} is-${iconSize}`;
    return <i className={classType}></i>;
  }
}
