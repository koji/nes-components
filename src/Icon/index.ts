import { NesIcon } from './Icon';
import { NesSNSIcon } from './SNSIcon';
import { NesReactionIcon } from './ReactionIcon';
import { NesArt } from './PixelArts';

export { NesIcon, NesSNSIcon, NesReactionIcon, NesArt };
