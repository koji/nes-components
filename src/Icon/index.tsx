import { NesIcon } from "./Icon";
import { NesSNSIcon } from "./SNSIcon";
import { NesReactionIcon } from "./ReactionIcon";

export { NesIcon, NesSNSIcon, NesReactionIcon };