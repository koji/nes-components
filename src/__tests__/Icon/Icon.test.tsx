import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { NesIcon, NesSNSIcon, NesReactionIcon, NesArt } from '../../Icon';
// import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
describe('<NesIcon>', () => {
  it('renders without crashing closeIcon', () => {
    const closeMediumIcon = renderer
      .create(<NesIcon iconType={'close'} iconSize={'medium'} />)
      .toJSON();
    expect(closeMediumIcon).toMatchSnapshot();
    const closeLargeIcon = renderer
      .create(<NesIcon iconType={'close'} iconSize={'large'} />)
      .toJSON();
    expect(closeLargeIcon).toMatchSnapshot();
    const closeSmallIcon = renderer
      .create(<NesIcon iconType={'close'} iconSize={'small'} />)
      .toJSON();
    expect(closeSmallIcon).toMatchSnapshot();
  });

  it('renders without crashing trophyIcon', () => {
    const closeMediumIcon = renderer
      .create(<NesIcon iconType={'trophy'} iconSize={'medium'} />)
      .toJSON();
    expect(closeMediumIcon).toMatchSnapshot();
    const closeLargeIcon = renderer
      .create(<NesIcon iconType={'trophy'} iconSize={'large'} />)
      .toJSON();
    expect(closeLargeIcon).toMatchSnapshot();
    const closeSmallIcon = renderer
      .create(<NesIcon iconType={'trophy'} iconSize={'small'} />)
      .toJSON();
    expect(closeSmallIcon).toMatchSnapshot();
  });

  it('renders without crashing coinIcon', () => {
    const closeMediumIcon = renderer
      .create(<NesIcon iconType={'coin'} iconSize={'medium'} />)
      .toJSON();
    expect(closeMediumIcon).toMatchSnapshot();
    const closeLargeIcon = renderer
      .create(<NesIcon iconType={'coin'} iconSize={'large'} />)
      .toJSON();
    expect(closeLargeIcon).toMatchSnapshot();
    const closeSmallIcon = renderer
      .create(<NesIcon iconType={'coin'} iconSize={'small'} />)
      .toJSON();
    expect(closeSmallIcon).toMatchSnapshot();
  });
});

describe('<NesArt>', () => {
  const nesLogo = renderer.create(<NesArt iconType={'nes-logo'} />).toJSON();
  expect(nesLogo).toMatchSnapshot();

  const snesLogo = renderer.create(<NesArt iconType={'snes-logo'} />).toJSON();
  expect(snesLogo).toMatchSnapshot();

  const nesOctcat = renderer
    .create(<NesArt iconType={'nes-octocat'} isAnimated={true} />)
    .toJSON();
  expect(nesOctcat).toMatchSnapshot();

  const nesSmartphone = renderer
    .create(<NesArt iconType={'nes-smartphone'} />)
    .toJSON();
  expect(nesSmartphone).toMatchSnapshot();
});

describe('<NesReactionIcon>', () => {
  const heartIcon = renderer
    .create(<NesReactionIcon iconType={'heart'} iconSize={'medium'} />)
    .toJSON();
  expect(heartIcon).toMatchSnapshot();

  const starIcon = renderer
    .create(
      <NesReactionIcon
        iconType={'star'}
        iconSize={'large'}
        iconReactionType={'half'}
      />
    )
    .toJSON();
  expect(starIcon).toMatchSnapshot();

  const likeIcon = renderer
    .create(
      <NesReactionIcon
        iconType={'like'}
        iconSize={'small'}
        iconReactionType={'transparent'}
      />
    )
    .toJSON();
  expect(likeIcon).toMatchSnapshot();
});

describe('<NesSNSIcon>', () => {
  const twitterIcon = renderer
    .create(<NesSNSIcon iconType={'twitter'} iconSize={'small'} />)
    .toJSON();
  expect(twitterIcon).toMatchSnapshot();

  const googleIcon = renderer
    .create(<NesSNSIcon iconType={'google'} iconSize={'large'} />)
    .toJSON();
  expect(googleIcon).toMatchSnapshot();

  const facebookIcon = renderer
    .create(<NesSNSIcon iconType={'facebook'} iconSize={'medium'} />)
    .toJSON();
  expect(facebookIcon).toMatchSnapshot();
});
