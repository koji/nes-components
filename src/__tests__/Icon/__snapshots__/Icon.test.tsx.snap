// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<i
  className="nes-logo"
/>
`;

exports[` 2`] = `
<i
  className="snes-logo"
/>
`;

exports[` 3`] = `
<i
  className="nes-octocat animate"
/>
`;

exports[` 4`] = `
<i
  className="nes-smartphone"
/>
`;

exports[` 5`] = `
<i
  className="nes-icon is-medium heart is-undefined"
/>
`;

exports[` 6`] = `
<i
  className="nes-icon is-large star is-half"
/>
`;

exports[` 7`] = `
<i
  className="nes-icon is-small like is-transparent"
/>
`;

exports[` 8`] = `
<i
  className="nes-icon undefined is-small"
/>
`;

exports[` 9`] = `
<i
  className="nes-icon undefined is-large"
/>
`;

exports[` 10`] = `
<i
  className="nes-icon undefined is-medium"
/>
`;

exports[`<NesIcon> renders without crashing 1`] = `
<i
  className="nes-icon close is-medium"
/>
`;

exports[`<NesIcon> renders without crashing 2`] = `
<i
  className="nes-icon close is-large"
/>
`;

exports[`<NesIcon> renders without crashing 3`] = `
<i
  className="nes-icon close is-small"
/>
`;

exports[`<NesIcon> renders without crashing closeIcon 1`] = `
<i
  className="nes-icon close is-medium"
/>
`;

exports[`<NesIcon> renders without crashing closeIcon 2`] = `
<i
  className="nes-icon close is-large"
/>
`;

exports[`<NesIcon> renders without crashing closeIcon 3`] = `
<i
  className="nes-icon close is-small"
/>
`;

exports[`<NesIcon> renders without crashing coinIcon 1`] = `
<i
  className="nes-icon coin is-medium"
/>
`;

exports[`<NesIcon> renders without crashing coinIcon 2`] = `
<i
  className="nes-icon coin is-large"
/>
`;

exports[`<NesIcon> renders without crashing coinIcon 3`] = `
<i
  className="nes-icon coin is-small"
/>
`;

exports[`<NesIcon> renders without crashing trophyIcon 1`] = `
<i
  className="nes-icon trophy is-medium"
/>
`;

exports[`<NesIcon> renders without crashing trophyIcon 2`] = `
<i
  className="nes-icon trophy is-large"
/>
`;

exports[`<NesIcon> renders without crashing trophyIcon 3`] = `
<i
  className="nes-icon trophy is-small"
/>
`;
